src/OSPSuite.Infrastructure.Import/Services/Importer.cs

```diff
@@ -210,11 +214,13 @@ public MappingProblem CheckWhetherAllDataColumnsAreMapped(IReadOnlyList<ColumnIn
                }
             }
 
-            if (!string.IsNullOrEmpty(molecularWeightDescription))
+            if (!molecularWeightValueAsString.IsNullOrEmpty())
             {
-               if (double.TryParse(molecularWeightDescription, out var molWeight))
+               if (double.TryParse(molecularWeightValueAsString, out var molWeight))
                {
-                  dataRepo.AllButBaseGrid().Each(x => x.DataInfo.MolWeight = molWeight);
+                  //we are assuming that the MW coming from the column in excel is always in g/mol, that's why we need this conversion here, 
+                  //otherwise it would be saved in kg/μmol
```

Comment:
The otherwise is weird. 
suggestion
```
 //we are assuming that the MW coming from the column in excel is always in g/mol, that's why we need the conversion to core unit to ensure that the value is saved properly
```